.circleci/config.yml

```diff
@@ -124,41 +118,50 @@ jobs:
           name: Lint Terraform code
           command: make lint-terraform
 
-  build-frontend:
+  build-and-deploy-frontend:
     docker:
       - image: cimg/node:22.10
     steps:
       - checkout
+      - aws-cli/setup:
+          role_arn: "$AWS_ROLE_ARN"
       - run:
           name: Install dependencies
           command: |
             cd frontend
             npm install
       - run:
-          name: build frontend
+          name: Prepare cache restoring
           command: |
-            cd frontend
-            npm run build
-            mv out build
-      - persist_to_workspace:
-          root: ~/project
-          paths:
-            - frontend/build
-
-  deploy-frontend:
-    docker:
-      - image: cimg/node:22.10
-    steps:
-      - checkout
-      - attach_workspace:
-          at: frontend/build
-      - aws-cli/setup:
-          role_arn: "$AWS_ROLE_ARN"
+            cd ~/project
+            PREVIOUS_COMMIT=$(git rev-parse HEAD^)
+            echo "Saving previous commit hash:"
+            echo "$PREVIOUS_COMMIT" | tee cachekey
+      - restore_cache:
+          keys:
+            - frontend-cache-{{ .Branch }}-{{ checksum "./cachekey" }}
+            - frontend-cache-main-{{ checksum "./cachekey" }}
       - run:
-          name: deploy frontend
+          name: build frontend
           command: |
             cd frontend
-            aws s3 sync build s3://pnp-character-application-frontend --region eu-central-1
+
+            chmod +x build_on_change.sh
+            ./build_on_change.sh
```

Comment:
Du kannst die Datei direkt in Git commiten mit der x-Flag gesetzt. Dann spart man sich den Befehl hier und kann es auch direct lokal ausführen